warrior/WarriorCore/common_execution_utils.py

```diff
@@ -12,8 +12,76 @@
 '''
 
 """Module that contains common utilities required for execution """
+import copy
+
 from Framework.Utils.print_Utils import print_warning
+import Framework.Utils as Utils
+
+def get_step_list(filepath, step_tag, sub_step_tag):
+    """
+    Takes the location of Testcase /Suite/Project xml file as input
+    finds a list of all the step elements present in the file
+    Parse the step object,
+    1. Retrieve the runmode and it corresponding value.
+    Based on runmode and value append the list.
+    2. Retrieve the retry and it corresponding value.
+    Based on runmode and value append the list.
+
+    :Arguments:
+        1. filepath    = full path of the Testcase/suite/project xml file
+        2. step_tag = xml tag for group of step in the file
+        3. sub_step_tag = xml tag for each step in the file
 
+    """
+    step_list_with_rmt_retry = []
+    root = Utils.xml_Utils.getRoot(filepath)
+    steps = root.find(step_tag)
+    if steps is None:
+        print_warning("The file: '{}' has no steps "
+                      "to be executed".format(filepath))
+    else:
+        step_list = steps.findall(sub_step_tag)
+        # iterate all steps to get the runmode and retry details
+        for index, step in enumerate(step_list):
+            runmode, value = get_runmode_from_xmlfile(step)
+            retry_type, _, _, retry_value, _ = get_retry_from_xmlfile(step)
+            if runmode is not None and value > 0:
+                # more than one step in step list, insert new step
+                if len(step_list) > 1:
+                    go_next = len(step_list_with_rmt_retry) + value + 1
+                    for i in range(0, value):
+                        copy_step = copy.deepcopy(step)
+                        copy_step.find("runmode").set("value", go_next)
+                        copy_step.find("runmode").set("attempt", i + 1)
+                        step_list_with_rmt_retry.append(copy_step)
+                # only one step in step list, append new step
+                else:
+                    for i in range(0, value):
+                        copy_step = copy.deepcopy(step)
+                        copy_step.find("runmode").set("attempt", i + 1)
```

Comment:
I know I wrote this more than a year ago...but I think this part also need to set the runmode value to go_next?